uuid/v7.ts

```diff
@@ -0,0 +1,115 @@
+ *
+ * @param id UUID value.
+ *
+ * @returns `true` if the string is a valid UUIDv7, otherwise `false`.
+ *
+ * @example Usage
+ * ```ts
+ * import { validate } from "@std/uuid/v7";
+ * import { assert, assertFalse } from "@std/assert";
+ *
+ * assert(validate("017f22e2-79b0-7cc3-98c4-dc0c0c07398f"));
+ * assertFalse(validate("fac8c1e0-ad1a-4204-a0d0-8126ae84495d"));
+ * ```
+ */
+export function validate(id: string): boolean {
+  return UUID_RE.test(id);
+}
+
+/**
+ * Options for {@linkcode generate}.
+ *
+ * @experimental **UNSTABLE**: New API, yet to be vetted.
+ */
+export interface GenerateOptions {
+  /**
+   * Unix epoch timestamp in milliseconds.
+   *
+   * @default {Date.now()}
+   */
+  timestamp?: number;
+}
+
+/**
+ * Generates a {@link https://www.rfc-editor.org/rfc/rfc9562.html#section-5.7 | UUIDv7}.
+ *
+ * @experimental **UNSTABLE**: New API, yet to be vetted.
+ *
+ * @throws {RangeError} If the timestamp is not a non-negative integer.
+ *
+ * @param options Options for generating the UUIDv7.
+ * @returns Returns a UUIDv7 string
+ *
+ * @example Usage
+ * ```ts
+ * import { generate, validate } from "@std/uuid/v7";
+ * import { assert } from "@std/assert";
+ *
+ * const uuid = generate();
+ * assert(validate(uuid));
+ * ```
+ */
+export function generate(options: GenerateOptions = {}): string {
+  const bytes = new Uint8Array(16);
+  const view = new DataView(bytes.buffer);
+  // Unix timestamp in milliseconds (truncated to 48 bits)
+  if (
+    options.timestamp !== undefined && (
+      !Number.isInteger(options.timestamp) || options.timestamp < 0
+    )
+  ) {
+    throw new RangeError(
+      `Cannot generate UUID as timestamp must be a non-negative integer: timestamp ${options.timestamp}`,
+    );
+  }
+  const timestamp = BigInt(options.timestamp ?? Date.now());
+  view.setBigUint64(0, timestamp << 16n);
+  crypto.getRandomValues(bytes.subarray(6));
+  // Version (4 bits) Occupies bits 48 through 51 of octet 6.
+  view.setUint8(6, (view.getUint8(6) & 0b00001111) | 0b01110000);
+  // Variant (2 bits) Occupies bits 64 through 65 of octet 8.
+  view.setUint8(8, (view.getUint8(8) & 0b00111111) | 0b10000000);
+  return bytesToUuid(bytes);
+}
+
+/**
+ * Extracts the timestamp from a UUIDv7.
+ *
+ * @experimental **UNSTABLE**: New API, yet to be vetted.
+ *
+ * @param uuid UUIDv7 string to extract the timestamp from.
+ * @returns Returns the timestamp in milliseconds.
+ *
+ * @throws {TypeError} If the UUID is not a valid UUIDv7.
+ *
+ * @example Usage
+ * ```ts
+ * import { extractTimestamp } from "@std/uuid/v7";
+ * import { assertEquals } from "@std/assert";
+ *
+ * const uuid = "017f22e2-79b0-7cc3-98c4-dc0c0c07398f";
+ * const timestamp = extractTimestamp(uuid);
+ * assertEquals(timestamp, 1645557742000);
+ * ```
+ */
+export function extractTimestamp(uuid: string): number {
+  if (!validate(uuid)) {
+    throw new TypeError(
+      "Could not extract timestamp because the UUID was not a valid UUIDv7",
```

Comment:
We have [the style guide](https://github.com/denoland/std/blob/main/.github/CONTRIBUTING.md#error-messages) of the error messages. I think the below form follows the guide better:

```suggestion
      `Cannot extract timestamp because the UUID (${uuid}) is not a valid UUIDv7`,
```

Reply:
How about the following?

```suggestion
     `Cannot extract timestamp because the UUID is not a valid UUIDv7: uuid is "${uuid}"`
```

Reply:
Update [here](https://github.com/denoland/std/pull/5887/commits/38c23954fd64806dec09f46dec216cfef3ffc3f1)